book/src/gobject_properties.md

```diff
@@ -1,35 +1,48 @@
 # Properties
 
-Properties allow us to access state of GObjects.
+Properties provide a public API for accessing state of GObjects.
 
 Let us see how this is done by experimenting with the `Switch` widget.
 One of its properties is the [state](https://docs.gtk.org/gtk4/property.Switch.state.html).
-It can be read and be written to.
-We do that by calling the `get_property` and `set_property` methods.
-Since all of this is highly dynamic, checks occur at runtime and the code involves a bit of boilerplate.
+According to the GTK docs, it can be read and be written to.
+That is why, `gtk-rs` provides corresponding [`get_state`](../docs/gtk4/struct.Switch.html#method.get_state) and [`set_state`](../docs/gtk4/docs/gtk4/struct.Switch.html#method.set_state) methods.
 
 <span class="filename">Filename: listings/gobject_properties/1/main.rs</span>
 
 ```rust ,no_run
 {{#rustdoc_include ../listings/gobject_properties/1/main.rs:switch}}
 ```
+Alternatively, we can use the general [`get_property`](http://gtk-rs.org/docs/glib/object/trait.ObjectExt.html#tymethod.get_property) and [`set_property`](http://gtk-rs.org/docs/glib/object/trait.ObjectExt.html#tymethod.set_property) methods.
+Because they can be used for properties of different types, they operate with the polymorphic [`glib::Value`](http://gtk-rs.org/docs/glib/value/struct.Value.html).
+The underlying type of `glib::Value` can be a boolean, integer, float or a GObject.
+It can even be optional and corresponds then to `Option<T>`.
```

Comment:
Only for some types though: all types that are nullable, so objects, strings, boxed types